pyop2/host.py

```diff
@@ -651,6 +645,14 @@ def compile(self, argtypes=None, restype=None):
         strip = lambda code: '\n'.join([l for l in code.splitlines()
                                         if l.strip() and l.strip() != ';'])
 
+        # Attach semantical information to the kernel's AST
+        if self._kernel._ast:
+            fundecl = ast_get_fun_decls(self._kernel._ast)
+            if fundecl:
+                for arg, f_arg in zip(self._args, fundecl.args):
+                    if arg._uses_itspace and arg._is_INC:
+                        f_arg.pragma = ast.WRITE
```

Comment:
I've changed this piece a bit in another branch I have, so I'll comment it properly there once "ready"